ixwebsocket/IXSocketOpenSSL.cpp

```diff
@@ -16,17 +16,23 @@
 #include <vector>
 #ifdef _WIN32
 #include <shlwapi.h>
+//To avoid the problem with X509_NAME collision between macro in wincrypt.h and x509v3.h
+//It has to be undef before x509v3.h and after wincrypt.h
+#undef X509_NAME
 #else
 #include <fnmatch.h>
 #endif
-#if OPENSSL_VERSION_NUMBER < 0x10100000L
+
```

Comment:
I'm pretty positive that old openssl installation will not have this x509v3.h header, so we should keep this OPENSSL_VERSION_NUMBER check or we will lose compatibility with old openssl.

Reply:
I am just a little bit confused here.  They way I read this is "`if version < 0X10100000L`" we **do** include `x509v3.h`.  They way you explaining sound like the opposite, with the line I removed,  `x509v3.h` is not included.  But then, I would expect this instead "`if version > 0X10100000L`"

I am using OpenSSL 3.0.7, and if I keep the line above, I don't have access to define from  `x509v3.h`.  So, I would suggest this : 
```
#if OPENSSL_VERSION_NUMBER > 0x10100000L
#include "openssl/x509v3.h"
#endif
```
So, the include file will be there only on more recent version of OpenSSL.